src/convertor/OndcConvertor.js

```diff
@@ -0,0 +1,69 @@
+import jqUtility from '../utility/JqUtility';
+import LoggingService from '../utility/LoggingService';
+
+const logger = LoggingService.getLogger('OndcConvertor');
+
+export default class OndcConvertor {
+  static generateOndcFilter(ondcMatchedValue, ondcDataType, platformTagValue) {
+    logger.debug(`ondcMatchedValue: ${JSON.stringify(ondcMatchedValue)}`);
+    logger.debug(`ondcDataType: ${ondcDataType}`);
+    logger.debug(`platformTagValue: ${platformTagValue}`);
+    const ondcTagKeys = ondcMatchedValue.ondc.split('.');
+    const ondcItemValue = `${platformTagValue} ${ondcDataType}`;
+    const ondcFilter = `. | if ( ${ondcTagKeys.length} > 1) 
+        then { ${ondcTagKeys[0]}: { ${ondcTagKeys[1]}: ${ondcItemValue} }}
+        else { ${ondcTagKeys[0]} : ${ondcItemValue} } end`;
+    logger.debug(`ondcFilter: ${ondcFilter}`);
+    return ondcFilter;
+  }
+
+  static generateOndcDataType(ondcMatchedValue) {
+    return ((ondcMatchedValue.ondcDataType === 'boolean' || ondcMatchedValue.ondcDataType === '' || ondcMatchedValue.platformValue === '')
+      ? '' : `| to${ondcMatchedValue.ondcDataType}`);
```

Comment:
move condition into separate variable , its very difficult to read in the ternary operator

Reply:
updated